File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini=prices[0];
        int maxip=0;
        int n=prices.length;
        for(int i=0;i<n;i++)
        {
            mini=Math.min(prices[i],mini);
            maxip=Math.max(prices[i]-mini,maxip);
        }
        return maxip;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max=0,min=prices[0];
        for(int i=1;i<prices.size();i++)
        {
            max=std::max(max,(prices[i]-min));
            min=std::min(prices[i],min);

        }
        
        return max;
    }
};